### PR TITLE
added arch diagram for rollup service

### PR DESCRIPTION
**Pull request by GeorgeJahad**

### CODE REVIEW

File: docs/arch.plantuml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+240m Tumbling Window
+==========
+1440m Tumbling Window
+] 
+note right: com.rackspacecloud.metrics.rollup.config.TopologyConfiguration
+card Rollups
+note right: com.rackspacecloud.metrics.rollup.domain.RolledUp
+note bottom: rollups contain: min, max, mean, sum, count for the corresponding window
+queue 5m
+note left: these are all kafka topics
```
Comment (itzg) [inline docs/arch.plantuml]: Perhaps note that the topic names are prefixed with "unified.metrics.json."

File: docs/arch.plantuml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+@startuml
+center header <b>Ceres Rollup Service Architecture
+queue "unified.metrics.json topic" as input
+card Metrics
+note right: com.rackspacecloud.metrics.rollup.domain.Metric
```
Comment (itzg) [inline docs/arch.plantuml:5]: Wow, I wish the standard ExternalMetric had been used here and then wrapper types for the accumulation logic.

File: docs/arch.plantuml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+==========
+60m Tumbling Window
+==========
+240m Tumbling Window
+==========
+1440m Tumbling Window
+] 
+note right: com.rackspacecloud.metrics.rollup.config.TopologyConfiguration
+card Rollups
+note right: com.rackspacecloud.metrics.rollup.domain.RolledUp
```
Comment (itzg) [inline docs/arch.plantuml:21]: ...and similarly, I would have expected the use of ExternalMetric but with special field suffixes, like "_max", "_avg".

**Review by itzg — COMMENTED** [review]
Thanks! You know I like diagrams :) Just the output topic clarification.

I should write up issues for the desired use of ExternalMetric.

**Comment by itzg** [comment]
Created https://jira.rax.io/browse/CERES-306

File: docs/arch.plantuml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+note bottom: rollups contain: min, max, mean, sum, count for the corresponding window
+queue 5m
+note left: these are all kafka topics
+queue 20m
+queue 60m
+queue 240m
+queue 1440m
+
+input --> Metrics
+Metrics --> KTable : grouped by key:  accountType.account.monitoringSystem.collectionName.device.deviceLabel.collectionLabel
```
Comment (jjbuchan) [inline docs/arch.plantuml:31]: I can't remember what each of these values would be but `deviceLabel` sounds worrying.

**Review by jjbuchan — DISMISSED** [review]

File: docs/arch.plantuml (hunk 4 above)
Comment (GeorgeJahad) [reply docs/arch.plantuml:31]: do we need to create a jira ticket to fix that?

Comment (itzg) [reply docs/arch.plantuml:31]: That seems like a good idea

Comment (GeorgeJahad) [reply docs/arch.plantuml:31]: I added this ticket for it:
https://jira.rax.io/browse/CERES-317

**Review by itzg — APPROVED** [review]

**Pull request merged**